comtypes/test/test_dispifc_safearrays.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+
+import unittest
+import comtypes
+
+from comtypes import CLSCTX_LOCAL_SERVER
+from comtypes.client import CreateObject, GetModule
+from ctypes import byref, pointer, c_double
+
+ComtypesCppTestSrvLib_GUID = "{07D2AEE5-1DF8-4D2C-953A-554ADFD25F99}"
+
+try:
+    GetModule((ComtypesCppTestSrvLib_GUID, 1, 0, 0))
+    import comtypes.gen.ComtypesCppTestSrvLib as ComtypesCppTestSrvLib
+
+    IMPORT_FAILED = False
+except (ImportError, OSError):
+    IMPORT_FAILED = True
+
+
+@unittest.skipIf(IMPORT_FAILED, "This depends on the out of process COM-server.")
+class Test_DispMethods(unittest.TestCase):
+    """Test dispmethods with safearray and safearray pointer parameters."""
+
+    UNPACKED_ZERO_VALS = tuple(0.0 for _ in range(10))
+    UNPACKED_CONSECUTIVE_VALS = tuple(float(i) for i in range(10))
+
+    def _create_dispifc(self) -> "ComtypesCppTestSrvLib.IDispSafearrayParamTest":
+        # Explicitely ask for the dispinterface of the component.
+        return CreateObject(
+            "Comtypes.DispSafearrayParamTest",
+            clsctx=CLSCTX_LOCAL_SERVER,
+            interface=ComtypesCppTestSrvLib.IDispSafearrayParamTest,
+        )
+
+    def _create_zero_array(self) -> comtypes.safearray._midlSAFEARRAY:
+        return comtypes.safearray._midlSAFEARRAY(c_double).create(
+            [c_double() for _ in range(10)]
+        )
+
+    def _create_consecutive_array(self) -> comtypes.safearray._midlSAFEARRAY:
+        return comtypes.safearray._midlSAFEARRAY(c_double).create(
+            [c_double(i) for i in range(10)]
+        )
+
+    def test_inout_byref(self):
+        dispifc = self._create_dispifc()
+        # Passing a safearray by reference to a method that has declared the parameter
+        # as [in, out] we expect modifications of the safearray on the server side to
+        # also change the safearray on the client side.
+        test_array = self._create_zero_array()
+        self.assertEqual(test_array.unpack(), self.UNPACKED_ZERO_VALS)
+        dispifc.InitArray(byref(test_array))
+        self.assertEqual(test_array.unpack(), self.UNPACKED_CONSECUTIVE_VALS)
+
+    def test_inout_pointer(self):
+        dispifc = self._create_dispifc()
+        # Passing a safearray pointer to a method that has declared the parameter
+        # as [in, out] we expect modifications of the safearray on the server side to
+        # also change the safearray on the client side.
+        test_array = self._create_zero_array()
+        self.assertEqual(test_array.unpack(), self.UNPACKED_ZERO_VALS)
+        dispifc.InitArray(byref(test_array))
```

Comment:
I apologize for confusing `test_inout_byref` and `test_inout_pointer`.

For this line, we should revert it to be the same as [the codebase when the test was added](https://github.com/enthought/comtypes/blob/ed89154a5d3cfa87434c92f6ca864ea972f152b5/comtypes/test/test_dispifc_safearrays.py#L53) in ed89154.

```suggestion
        dispifc.InitArray(pointer(test_array))
```

Reply:
Ooops, sorry I missed that when looking at your changes.